lib/src/lifecycle_module.dart

```diff
@@ -83,23 +107,45 @@ abstract class LifecycleModule {
         new StreamController<LifecycleModule>.broadcast();
     _didUnloadChildModuleController =
         new StreamController<LifecycleModule>.broadcast();
+    _willSuspendController = new StreamController<LifecycleModule>.broadcast();
+    _didSuspendController = new StreamController<LifecycleModule>.broadcast();
+    _willResumeController = new StreamController<LifecycleModule>.broadcast();
+    _didResumeController = new StreamController<LifecycleModule>.broadcast();
+
+    _logger = new Logger(name);
   }
 
+  /// Whether the module is currently loaded.
+  bool get isLoaded => _isLoaded;
+
+  /// Whether the module is currently suspended.
+  ///
+  /// This will always be false when the module is not loaded.
+  bool get isSuspended => _isSuspended;
+
   //--------------------------------------------------------
   // Public methods that can be used directly to trigger
   // module lifecycle / check current lifecycle state
   //--------------------------------------------------------
 
   /// Public method to trigger the loading of a Module.
+  ///
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
+  /// TODO: Do we want to support re-loading of child modules here?
   Future load() {
     Completer completer = new Completer();
-    _willLoadController.add(this);
-    onLoad().then((_) {
-      _didLoadController.add(this);
+    if (!_isLoaded) {
+      _willLoadController.add(this);
+      onLoad().then((_) {
+        _didLoadController.add(this);
+        _isLoaded = true;
+        completer.complete();
+      });
+    } else {
+      _logger.warning('Module is already loaded, cannot load.');
```

Comment:
Oh, didn't even notice that you were constructing the module with the `name` property. But like Max commented above it would be nice to have `w_module` in there as well